src/main/java/com/labsynch/labseer/domain/Parent.java

```diff
@@ -710,6 +712,37 @@ public static TypedQuery<Parent> findParentsBySaltForms(Set<SaltForm> saltForms,
         return q;
     }
 
+    public static TypedQuery<Parent> checkForDuplicateParentByCdId(Long parentId, int[] cdIds) {
+        // Given a parent ID and list of structure ids
+        // Return a list of duplicates which have theame stereo category and stereo comment as the parent id
+        // exclude the parent id from the list
+        if(cdIds == null)
+            throw new IllegalArgumentException("The cdIds argument is required");
+        if(parentId == null)
+            throw new IllegalArgumentException("The parentId argument is required");
+
+        Parent parent = findParent(parentId);
+        if(parent == null)
+            throw new IllegalArgumentException("The parentId argument is invalid - no parent with id " + parentId + " found");
+        
+        EntityManager em = Parent.entityManager();
+        String queryBuilder = "SELECT o FROM Parent AS o WHERE o.CdId IN (:cdIds) AND o.id != :id and o.stereoCategory = :stereoCategory";
+
+        if(parent.getStereoComment() == null) {
+            queryBuilder += " AND o.stereoComment IS NULL";
+        } else {
+            queryBuilder += " AND o.stereoComment = :stereoComment";
```

Comment:
This is not a ignore case match like it is in the code I am replacing. This will be fixed in an upcoming pull request.